snappy/install.go

```diff
@@ -268,10 +257,15 @@ func GarbageCollect(name string, flags InstallFlags, pb progress.Meter) error {
 		return nil
 	}
 
-	for _, snap := range snaps[:active-1] {
+	for i, snap := range snaps {
```

Comment:
If I'm reading this correct we remove all except the currently active snap. And this code runs after the new snap got installed. So there is only a single version installed afterwards(?). If that is the case we break rollback of broken OS/kernel updates here ? or am I missing something and I misremember the order?